datafusion/sqllogictest/test_files/cte.slt

```diff
@@ -19,3 +19,195 @@ query II
 select * from (WITH source AS (select 1 as e) SELECT * FROM source) t1,   (WITH source AS (select 1 as e) SELECT * FROM source) t2
 ----
 1 1
+
```

Comment:
Could you also add some tests that do:

1. EXPLAIN explain plans of these queries to show how they work
2. Tests that produce more than one batch on each iteration (these tests make batches much smaller than the default 8K rows). Maybe by setting `target_batch_size` to `3` or something.
3. A query with some more stateful exeution (e.g. a GROUP BY)

Reply:
sounds great. Will do. We've been using it in our use case where we nest regular CTEs within each term of the recursive CTE, so I'll add those too.

Reply:
Added these tests!